File: src/Moryx.Runtime.Maintenance/Plugins/Databases/Wcf/Models/DataModel.cs
// Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.Runtime.Maintenance.Plugins.Databases
{
    /// <summary>
    /// Conatins the data to the target model.
    /// </summary>
    [DataContract]
    public class DataModel
    {
        /// <summary>
        /// Name of the target model..
        /// </summary>
        [DataMember]
        public string TargetModel { get; set; }

        /// <summary>
        /// Configuration of the database model.
        /// </summary>
        [DataMember]
        public DatabaseConfigModel Config { get; set; }

        /// <summary>
        /// An amount of setups for this model.
        /// </summary>
        [DataMember]
        public SetupModel[] Setups { get; set; }

        /// <summary>
        /// An amount of backups of this model.
        /// </summary>
        [DataMember]
        public BackupModel[] Backups { get; set; }

        /// <summary>
        /// Available migrations for this context
        /// </summary>
        [DataMember]
        public DbMigrationsModel[] AvailableMigrations { get; set; }

        /// <summary>
        /// Installed migrations for this context
        /// </summary>
        [DataMember]
        public DbMigrationsModel[] AppliedMigrations { get; set; }
    }
}

File: src/Moryx.Maintenance.Web/src/databases/models/DataModel.ts
/*
 * Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

import BackupModel from "./BackupModel";
import DatabaseConfigModel from "./DatabaseConfigModel";
import DbMigrationsModel from "./DbMigrationsModel";
import SetupModel from "./SetupModel";

export default class DataModel {
    public TargetModel: string;
    public Config: DatabaseConfigModel;
    public Setups: SetupModel[];
    public Backups: BackupModel[];
    public AvailableMigrations: DbMigrationsModel[];
    public AppliedMigrations: DbMigrationsModel[];
}
